src/dll.py

```diff
@@ -0,0 +1,92 @@
+"""Implementation of Doubly Linked List data type."""
+
+
+class DoublyLinkedList(object):
+    """Class representation of doubly linked list."""
+
+    def __init__(self, iterable=None):
+        """Instantiate linked list."""
+        self.head_node = None
+        self.tail_node = None
+        self.length = 0
```

Comment:
When you pass in an interable, are you getting the DLL you expect?
What happens if the value you pass in is a string? What if it is a single number?